Feature shortcuts dinamicos

sebalagos90 opened the pull request:
- Se crea el método para crear shortcuts dinámicos.
- Se crea el método para eliminar todos los shortcuts dinámicos.
- Se actualizan los archivos README.

sebalagos90 committed:
Merge branch 'master' into feature-shortcuts-dinamicos

sebalagos90 committed:
Se crea el método para crear shortcuts dinámicos.
Se considera que el ícono del shortcut debe ser base 64.

sebalagos90 committed:
Se implementa el método para eliminar todos los shortcuts dinámicos.

sebalagos90 committed:
Se cambia el nombre del método para crear shortcuts.

sebalagos90 committed:
Se actualizan los READMEs acorde a los últimos cambios del plugin.

xchopox commented on README.md:
```diff
@@ -4,13 +4,17 @@
+
+## Static shortcuts configuration
+
+Create android-shortcuts.json file in project's root folder with all the information about the shortcuts you want to create (maximum 4 shortcuts):
```
[S:0] Se sugiere dejar el comienzo de la frase como: "Create a JSON file called `android-shortcuts.json` in the root folder of the project..." o "Create the `android-shortcuts.json` file in the root folder of the project... ".

xchopox commented on README.md:
```diff
@@ -43,6 +47,37 @@ To improve launch performance you should set the following into config.xml file:
+
+### Create a dynamic shortcut
+
+This plugin allows you the following method to create shortcuts dynamically (on runtime):
```
[S:0] "This plugin allows you `to use` the following method..."

sebalagos90 replied:
My bad, tenía que haber sido "This plugin provides you ..."

xchopox commented on README.md:
```diff
@@ -43,6 +47,37 @@ To improve launch performance you should set the following into config.xml file:
+
+### Remove all dynamic shortcuts
+
+Use the following method to remove all dynamic shortcuts. This will not remove static shortcuts:
```
[S:0] "This will not remove `the` static shortcuts:"

xchopox commented on src/android/cl/kunder/androidshortcuts/AndroidShortcutsPlugin.java:
```diff
@@ -31,6 +38,34 @@ public boolean execute(String action, JSONArray args, final CallbackContext call
             }
+        } else if("createDynamicShortcut".equals(action)) {
+            ShortcutManager shortcutManager = this.cordova.getActivity().getSystemService(ShortcutManager.class);
+            if(shortcutManager.getDynamicShortcuts().size() == 4) {
```
[S:1] Favor de verificar condicional para los shortcuts estáticos.

xchopox reviewed (changes requested):
Verificar sugerencias especificadas en comentarios.

sebalagos90 committed:
Se realizan las correcciones del README.md y del contador de shortcuts total acorde a los comentarios del pull request.

sebalagos90 committed:
Se añaden los callbacks success faltantes.
Se corrigen los mensajes de error acorde al tipo de error.

sebalagos90 committed:
Cambios en seteo de iconos.

Se agrega fs-extra como dependencia del proyecto y se optimiza el hook para copiar los íconos.
Se modifica el archivo java del plugin para aceptar el ícono como base64 o string simple.

sebalagos90 committed:
Se realizan optimizaciones en el hook.

sebalagos90 committed:
Optimizaciones y actualización de READMEs.

Se realizan optimizaciones de código nativo.
Se actualiza la documentación acorde a los cambios efectuados en el script y en la funcionalidad.

xchopox reviewed (approved):
Cambios aprobados.
